server/server.go

```diff
@@ -16,19 +16,40 @@ import (
 	"github.com/joshprzybyszewski/cribbage/server/persistence"
 )
 
+type dbGetter func() (persistence.DB, error)
+
 type cribbageServer struct {
-	db persistence.DB
+	getDB dbGetter
 }
 
-func newCribbageServer(db persistence.DB) *cribbageServer {
+func newCribbageServer(getDB dbGetter) *cribbageServer {
 	return &cribbageServer{
-		db: db,
+		getDB: getDB,
 	}
 }
 
 func (cs *cribbageServer) NewRouter() http.Handler {
 	router := gin.Default()
 
+	router.LoadHTMLGlob("templates/*")
+	router.Static("/assets", "./assets")
+
+	router.GET("/ping", func(c *gin.Context) {
+		c.JSON(http.StatusOK, gin.H{
+			"message": "pong",
+		})
+	})
+
+	router.GET("/", handleIndex)
+
+	// Simple group: user. Used for serving pages affiliated with a given user
+	user := router.Group("/user")
+	{
+		user.GET("/", handleGetUser)
+		user.GET("/:username", cs.handleGetUsername)
+		user.GET("/:username/game/:gameID", cs.handleGetUsernameGame)
```

Comment:
we should convert to using the new "pass in a json object as the body" instead of changing our url:#

Reply:
nope